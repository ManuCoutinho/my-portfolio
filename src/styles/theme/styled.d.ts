import 'styled-components'

declare module 'styled-components' {
  export interface DefaultTheme extends CustomTheme {
    title: string
    colors: {
      black: string
      white: string
      background: string
      body: string
      primary: string
      secondary: string
      text: string
      title: string
      info: string
      highlight: string
      carousel_1: string
      carousel_2: string
      carousel_3: string
      carousel_4: string
      bg_card: string
      glass: string
    }
    font: {
      text: string
      title: string
    }
    fontSize: {
      xsm: string
      sm: string
      md: string
      lg: string
      xl: string
      xlg: string
    }
    fontWeight: {
      light: number
      normal: number
      semibold: number
      bold: number
    }
    transition: string
    radius: {
      small: string
      default: string
      medium: string
      full: string
    }
  }

  export interface CustomTheme {
    title: string
    colors: {
      black: string
      white: string
      background: string
      body: string
      primary: string
      secondary: string
      text: string
      title: string
      info: string
      highlight: string
      carousel_1: string
      carousel_2: string
      carousel_3: string
      carousel_4: string
      bg_card: string
      glass: string
    }
  }
}
