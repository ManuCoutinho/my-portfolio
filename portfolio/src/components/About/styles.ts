import styled from 'styled-components'

export const AboutSection = styled.section`
  width: 100%;
  padding: 0.5rem;
  margin: 0 auto;
`
