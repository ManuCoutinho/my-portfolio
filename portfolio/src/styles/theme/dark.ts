export const dark = {
  title: 'light',
  colors: {
    black: '#090B09',
    white: '#FFFCF2',
    background: '#1C221C',
    body: '#121713',
    primary: '#1A7444',
    secondary: '#12BF62',
    text: '#e8e4df',
    title: '#434a3c',
    info: '#AEAFAC',
    highlight: '#5AFF15',
  },
  font: {
    text: "'Montserrat Alternates', sans-serif",
    title: "'Josefin Sans', sans-serif",
  },
  fontSize: {
    xsm: '0.75em',
    sm: '0.875em',
    md: '1em',
    lg: '1.5em',
    xl: '2.5em',
    xlg: '3.5em',
  },
  fontWeight: {
    light: 100,
    normal: 400,
    semibold: 600,
    bold: 700,
  },
  transition: 'all 0.2s ease-in-out',
  radius: {
    default: '16px',
    medium: '25px',
    full: '50%',
  },
}
