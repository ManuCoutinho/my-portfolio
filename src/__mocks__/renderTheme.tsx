import { ReactElement, ReactNode, FC } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeContextProvider } from 'contexts/ThemeContext'

const MockProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <ThemeContextProvider>{children}</ThemeContextProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: MockProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
