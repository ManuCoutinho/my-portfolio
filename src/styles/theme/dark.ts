import { lighten } from 'polished'

const dark = {
  title: 'dark',
  colors: {
    black: '#090B09',
    white: '#FFFCF2',
    background: '#090B09',
    body: '#090B09',
    primary: '#1A7444',
    secondary: '#12BF62',
    text: '#DEE2E6',
    title: '#02C39A',
    info: '#78a590',
    highlight: 'rgba(15, 255, 103, 0.2)',
    active: '#37d27f',
    carousel_1: '#2D392D',
    carousel_2: '#080a08',
    carousel_3: 'rgba(10, 13, 10, 0.1)',
    carousel_4: 'rgba(255, 255, 255, 0.02)',
    bg_card: '#1F1F1F',
    glass: 'rgba(9, 11, 9, 0.65)',
    bg_modal: '#292929',
    disabled: lighten(0.5, '#1c1c1c'),
    border_1: 'rgba(15, 255, 103, 1)',
    border_2: '#37d27f'
  }
} as const

export default dark
