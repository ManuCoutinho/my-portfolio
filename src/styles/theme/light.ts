import { lighten } from 'polished'
const light = {
	title: 'light',
	colors: {
		black: '#353131',
		white: '#f3faf0',
		background: '#e48c59',
		body: '#f3faf0',
		primary: 'rgb(201, 100, 81)',
		secondary: '#ae6a40',
		text: '#003844',
		title: '#353131',
		info: '#8A5536',
		highlight: 'rgb(58, 12, 163, 0.5)',
		carousel_1: '#DEDDDB',
		carousel_2: '#f5f5f5',
		carousel_3: '#CBCAC7',
		carousel_4: '#f5f5f5',
		active: '#F05D51',
		bg_card: '#f8e3dd',
		glass: 'rgba(243, 250, 240, 0.1)',
		bg_modal: '#e19b76',
		disabled: lighten(0.5, '#56667c')
	}
} as const

export default light
