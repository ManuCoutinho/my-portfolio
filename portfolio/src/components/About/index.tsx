import { Container, Title } from '../Foundation/base'
import { AboutSection } from './styles'
export const About: React.FC = () => {
  return (
    <Container>
      <AboutSection>
        <Title>Sobre</Title>
      </AboutSection>
    </Container>
  )
}
