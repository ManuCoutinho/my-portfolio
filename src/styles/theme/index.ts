import { DefaultTheme, CustomTheme } from 'styled-components'
import dark from './dark'
import light from './light'

export type Theme = typeof defaultTheme

const defaultTheme = {
	font: {
		cursive:
			"'Montserrat Alternates', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif",
		regular:
			"'Josefin Sans', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, 'Opens Sans', 'Helvetica Neue', sans-serif"
	},
	fontSize: {
		xsm: '0.5rem', //8px
		sm: '0.875rem', //14px
		md: '1rem', //16px
		wd: '1.25rem', //20px
		lg: '1.5rem', //24px
		xl: '2rem', //36px
		xlg: '3rem' //48px
	},
	fontWeight: {
		thin: 100,
		light: 200,
		regular: 400,
		semibold: 600,
		bold: 700
	},
	radius: {
		sm: '8px',
		md: '16px',
		pill: '25px',
		full: '50%'
	}
} as const

function combineTheme(theme: CustomTheme): DefaultTheme {
	return { ...defaultTheme, ...theme }
}
export { combineTheme, dark, light }
