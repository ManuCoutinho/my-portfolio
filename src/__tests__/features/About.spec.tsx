import { screen } from '@testing-library/react'
import { render } from '../../styles/theme/renderTheme'
import About from '../../views/AboutSection'

describe('<About />', () => {
  it('should render About section components correctly', () => {
    render(<About />)
    expect(screen.getByRole('heading', { name: /sobre/i })).toBeInTheDocument()
    expect(
      screen.getByRole('img', { name: /girl coding with cup of coffee/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('definition')).toBeInTheDocument()
    expect(screen.getByTestId('slider')).toBeInTheDocument()
  })
  it('should match to snapshot', () => {
    const { container } = render(<About />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
