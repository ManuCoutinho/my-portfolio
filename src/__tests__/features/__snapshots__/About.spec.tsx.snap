// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should match to snapshot 1`] = `
<section
  class="sc-ba46d7a0-0 eMdtzG"
  id="about"
>
  <h2
    class="sc-776a3b16-0 ehhqFc"
  >
    Sobre
  </h2>
  <div
    class="sc-ba46d7a0-1 gNQsyi"
  >
    <div
      class="sc-ba46d7a0-2 goOiNi"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="girl coding with cup of coffee"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="sc-9a5cd56e-0 sc-f32d69d5-0 gjXRAA cOKkbS"
      direction="column"
      role="definition"
    >
      <h3
        class="sc-776a3b16-0 flLuIJ"
      >
        <span>
          Olá, sou
        </span>
         Manu Coutinho!
      </h3>
      <p
        class="sc-6c8a52e-0 gpITMB"
      >
        Eu sou uma desenvolvedora front-end, movida por uma paixão em construir coisas capazes de integrar diversos universos por meio da tecnologia.
      </p>
      <p
        class="sc-6c8a52e-0 gpITMB"
      >
        Gosto de desenvolver interfaces dinâmicas, de alta fidelidade e com designs pensados para transferir a melhor experiência ao usuário final.
      </p>
      <p
        class="sc-6c8a52e-0 gpITMB"
      >
        Minha abordagem para a resolução de problemas é altamente inclusiva, colaborativa e centrada no ser humano.
      </p>
      <p
        class="sc-6c8a52e-0 gpITMB"
      >
        Aqui, compartillho alguns dos meus projetos!
      </p>
    </div>
  </div>
  <div
    class="sc-3ba3f456-0 jrhHuJ"
  >
    <h4
      class="sc-776a3b16-0 flLuIJ"
    >
      Minhas Skills:
    </h4>
    <div
      class="sc-3ba3f456-1 jFhrOZ keen-slider"
      data-testid="slider"
    >
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="akar-icons:html-fill"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="mdi:jquery"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="fa-brands:react"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="fa-brands:sass"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="ion:logo-javascript"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="mdi:nodejs"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:storybook"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:sentry"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="mdi:graphql"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="ph:figma-logo"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="simple-icons:testinglibrary"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="fa-brands:github"
        />
      </div>
    </div>
    <div
      class="sc-3ba3f456-1 jFhrOZ keen-slider"
    >
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:typescript-alt"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:styledcomponents"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:nextjs"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:vite"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="simple-icons:jamstack"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:tailwind"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="simple-icons:mui"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="simple-icons:chakraui"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="file-icons:jest"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="fa-brands:bootstrap"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="akar-icons:css-fill"
        />
      </div>
      <div
        class="sc-3ba3f456-2 kzWrSk keen-slider__slide"
      >
        <iconify-icon
          class="sc-3ba3f456-3 fPTSnf"
          classname="sc-3ba3f456-3 fPTSnf"
          icon="fa-brands:git-alt"
        />
      </div>
    </div>
  </div>
</section>
`;
